### PR TITLE
Only deploy Airplane tasks that have been changed

### PR DESCRIPTION
**Pull request by leeweisberger**
The GitHub action adds significant load to our deploy/build system b/c every push to a repo deploys every Airplane task.

Update so that we diff the pushed HEAD with the latest commit on the branch and only deploy tasks that are affected by that diff.

In order to do this, I changed our action to a composite action which is an action that calls other actions. it:
1. Checks out the user's repo
2. Calculates changed files from the push
3. Calls the airplane deploy action (same as the action before this change!), passing in the changed files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +17,6 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
```
Comment (leeweisberger) [inline README.md:19]: This is no longer required. Our action calls this internally. It's still ok to have, but isn't necessary

File: deploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: "Airplane Deploy"
```
Comment (leeweisberger) [inline deploy/action.yml:1]: This is the same as the original ./action.yml

File: README.md [hunk 3]
```diff
@@ -28,9 +28,9 @@ jobs:
 
 ## Inputs
 
-| Name          | Requirement | Default | Description |
-| ------------- | ----------- | ------- | ----------- |
-| `api-key`     | **required**  | | An airplane API key. Generate by running `airplane apikeys create <key name>` from the [Airplane CLI](https://docs.airplane.dev/platform/airplane-cli)|
-| `team-id`  | **required**  | | Your airplane team ID. Find this in your [team settings](https://app.airplane.dev/settings/team)  |
-| `task-directory`   | _optional_  | `./` | The relative path from your GitHub repo root where your airplane tasks live. Setting this will speed up your deploy if your repo contains many files that aren't airplane tasks |
-| `env` | _optional_  | | The environment you want to deploy the tasks into. Defaults to your default environment.
+| Name             | Requirement  | Default | Description                                                                                                                                                                     |
```
Comment (leeweisberger) [inline README.md:31]: this was all just auto formatted

File: action.yml [hunk 4]
```diff
@@ -32,15 +32,32 @@ inputs:
...
+      with:
+        fetch-depth: 0
+    # Get all files changed from the last commit on the remote branch.
+    - id: files
+      uses: tj-actions/changed-files@v19
+      with:
+        separator: ","
+        since_last_remote_commit: true
+    # Write the CSV changed files into a newline separated file expected by the Airplane CLI.
+    - run: echo ${{ steps.files.outputs.all_changed_files }} | tr ',' '\n' > .airplane_changed_files.txt
```
Comment (francescaleung) [inline action.yml:48]: seems brittle but I guess using `\n` as a separator is a nonstarter

**Review by francescaleung — APPROVED** [review]

**Pull request merged**